pkg/ccl/importccl/import_stmt.go

```diff
@@ -696,67 +736,66 @@ func (r *importResumer) Resume(
 // stuff to delete the keys in the background.
 func (r *importResumer) OnFailOrCancel(ctx context.Context, txn *client.Txn) error {
 	details := r.job.Details().(jobspb.ImportDetails)
-	if details.BackupPath != "" {
-		return nil
-	}
 
 	// Needed to trigger the schema change manager.
 	if err := txn.SetSystemConfigTrigger(); err != nil {
 		return err
 	}
 	b := txn.NewBatch()
 	for _, tbl := range details.Tables {
-		tableDesc := tbl.Desc
-		tableDesc.State = sqlbase.TableDescriptor_DROP
-		// If the DropTime if set, a table uses RangeClear for fast data removal. This
-		// operation starts at DropTime + the GC TTL. If we used now() here, it would
-		// not clean up data until the TTL from the time of the error. Instead, use 1
-		// (that is, 1ns past the epoch) to allow this to be cleaned up as soon as
-		// possible. This is safe since the table data was never visible to users,
-		// and so we don't need to preserve MVCC semantics.
-		tableDesc.DropTime = 1
-		b.CPut(sqlbase.MakeDescMetadataKey(tableDesc.ID), sqlbase.WrapDescriptor(tableDesc), nil)
+		tableDesc := *tbl.Desc
+		tableDesc.Version++
+		if tbl.IsNew {
+			log.Infof(ctx, "dropping %q via cput  %v \n%v\n -> \n%v\n", tbl.Desc.Name, sqlbase.MakeDescMetadataKey(tableDesc.ID), sqlbase.WrapDescriptor(tbl.Desc), sqlbase.WrapDescriptor(&tableDesc))
+			tableDesc.State = sqlbase.TableDescriptor_DROP
+			// If the DropTime if set, a table uses RangeClear for fast data removal. This
+			// operation starts at DropTime + the GC TTL. If we used now() here, it would
+			// not clean up data until the TTL from the time of the error. Instead, use 1
+			// (that is, 1ns past the epoch) to allow this to be cleaned up as soon as
+			// possible. This is safe since the table data was never visible to users,
+			// and so we don't need to preserve MVCC semantics.
+			tableDesc.DropTime = 1
+			b.CPut(sqlbase.MakeNameMetadataKey(tableDesc.ParentID, tableDesc.Name), nil, tableDesc.ID)
+		} else {
```

Comment:
So this else case is for that not-yet-written import into existing tables feature?

Reply:
Correct